File: carrera de caballos/web/js/util/GameConfig.js
/**
 * Configuración del juego.
 */
const GameConfig = {
  DEFAULT_TRACK_LENGTH: 7,
  MIN_PLAYERS: 1,
  MAX_PLAYERS: 8,
  DECK_40: 40,
  DECK_48: 48,

  trackLength: 7,
  numPlayers: 2,
  deckSize: 40,
  betsEnabled: true,
  autoMode: false,

  getTrackLength() { return this.trackLength; },
  setTrackLength(v) { this.trackLength = Math.max(1, Math.min(20, v)); },

  getNumPlayers() { return this.numPlayers; },
  setNumPlayers(v) { this.numPlayers = Math.max(this.MIN_PLAYERS, Math.min(this.MAX_PLAYERS, v)); },

  getDeckSize() { return this.deckSize; },
  setDeckSize(v) { this.deckSize = v === this.DECK_48 ? this.DECK_48 : this.DECK_40; },

  isBetsEnabled() { return this.betsEnabled; },
  setBetsEnabled(v) { this.betsEnabled = !!v; },

  isAutoMode() { return this.autoMode; },
  setAutoMode(v) { this.autoMode = !!v; }
};

window.GameConfig = GameConfig;

File: carrera de caballos/src/main/java/carrera/util/GameConfig.java
package carrera.util;

/**
 * Configuración del juego: tamaño pista, jugadores, baraja 40/48, apuestas, modo auto/paso a paso.
 */
public class GameConfig {
    public static final int DEFAULT_TRACK_LENGTH = 7;
    public static final int MIN_PLAYERS = 1;
    public static final int MAX_PLAYERS = 8;
    public static final int DECK_40 = 40;
    public static final int DECK_48 = 48;

    private int trackLength;
    private int numPlayers;
    private int deckSize;
    private boolean betsEnabled;
    private boolean autoMode;

    public GameConfig() {
        this.trackLength = DEFAULT_TRACK_LENGTH;
        this.numPlayers = 2;
        this.deckSize = DECK_40;
        this.betsEnabled = true;
        this.autoMode = false;
    }

    public int getTrackLength() {
        return trackLength;
    }

    public void setTrackLength(int trackLength) {
        this.trackLength = Math.max(1, Math.min(20, trackLength));
    }

    public int getNumPlayers() {
        return numPlayers;
    }

    public void setNumPlayers(int numPlayers) {
        this.numPlayers = Math.max(MIN_PLAYERS, Math.min(MAX_PLAYERS, numPlayers));
    }

    public int getDeckSize() {
        return deckSize;
    }

    public void setDeckSize(int deckSize) {
        this.deckSize = deckSize == DECK_48 ? DECK_48 : DECK_40;
    }

    public boolean isBetsEnabled() {
        return betsEnabled;
    }

    public void setBetsEnabled(boolean betsEnabled) {
        this.betsEnabled = betsEnabled;
    }

    public boolean isAutoMode() {
        return autoMode;
    }

    public void setAutoMode(boolean autoMode) {
        this.autoMode = autoMode;
    }
}
